lib/google-apis/auth-client.js

```diff
@@ -159,218 +80,60 @@ AuthClient.prototype._generateErrorCreationUrl = function() {
- * not.
- * @function _checkStatusCodeRepeatability
- * @returns {Boolean} - Whether or not the status code grants retry
- */
-AuthClient.prototype._checkStatusCodeRepeatability =
-    function(statusCode) {
-
-  switch (statusCode) {
-  case 429:
-  case 500:
-  case 502:
-  case 503:
-    return true;
-  default:
-    return false;
-  }
-};
-
-/**
- * Checks the length of the `_requestQueue` property on the instance,
- * indicating whether or not another request or set of requests is still pending
- * execution.
- * @function _areRequestsStillInQueue
- * @returns {Number} - the length of the `_requestQueue` property
+ *  for submission to the service
+ * @param {RequestHandler~requestCallback} [userCb] - function called when the
+ *  request has succeeded or failed.
+ * @returns {Undefined} - does not return anything
  */
-AuthClient.prototype._areRequestsStillInQueue = function() {
-
-  return this._requestQueue.length > 0;
-};
+RequestHandler.prototype.sendError = function(errorMessage, userCb) {
 
-/**
- * Creates a timeout to wait for AuthClient init and calls back to the
- * queueNextRequest function to reattempt queueing the next request.
- * @function _pauseForClientInitialization
- * @returns {Undefined} does not return a value
- */
-AuthClient.prototype._pauseForClientInitialization = function() {
-
-  setTimeout(this._queueNextRequest.bind(this), 1500);
-};
-
-/**
- * Clears the request queue of any pending requests by reallocating the
- * property as a new array.
- * @function _clearRequestQueue
- * @returns {Undefined} does not return a value
- */
-AuthClient.prototype._clearRequestQueue = function() {
-
-  for (var i = 0; i < this._requestQueue.length; i++) {
-
-    if (isFunction(this._requestQueue[i][1])) {
-
-      this._requestQueue[i][1](this._initError, null, null);
-    }
-  }
+  var cb = isFunction(userCb) ? userCb : function () { return; };
 
-  this._requestQueue = [];
-};
-
-/**
- * Attempts to determine if the next queued request can be executed by checking
- * whether or not the client is already requesting or has initialized. If the
- * client is not requesting and the client has already initialized then the
- * next request in the queue can be executed.
- * @function _queueNextRequest
- * @returns {Undefined} - does not return a value
- */
-AuthClient.prototype._queueNextRequest = function() {
-
-  if (!isNull(this._initError)) {
-
-    this._clearRequestQueue();
-
-    return;
-  } else if (this._isRequesting) {
-
-    return;
-  } else if (!this._hasInited) {
+  if (!this._shouldReportErrors) {
 
-    this._pauseForClientInitialization();
+    cb(new Error([
+        'Client has not been configured to send errors, please check the',
+        'NODE_ENV enviorment variable and make sure it is set to production'
+      ].join(' ')), null, null);
 
     return;
   }
 
-  this._isRequesting = true;
-  this._makeRequest();
+  this._requestFactory(
+    {
+      url : this._generateErrorCreationUrl(),
```

Comment:
I think the URL is a constant once the RequestHandler is created. Could you compute and cache this in the constructor? A small function that is only called from the constructor should probably be inlined there.